### PR TITLE
Feature/e2e integration tests

### PR DESCRIPTION
**Pull request by knighto82**

### CODE REVIEW

File: src/main/java/io/github/jpmorganchase/fusion/Fusion.java [hunk 1]
```diff
@@ -603,6 +610,11 @@ public FusionBuilder rootURL(String rootURL) {
             this.rootURL = rootURL;
             return this;
         }
+
+        public FusionBuilder datasetBearerToken(String catalog, String dataset, String token) {
```
Comment (martinmcclintock) [inline src/main/java/io/github/jpmorganchase/fusion/Fusion.java:614]: Is this the right name for it? I'm not sure of a better one though - Fusion Download Token?

File: src/test/java/io/github/jpmorganchase/fusion/pact/util/FileHelper.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+import java.io.ByteArrayOutputStream;
+import java.io.InputStream;
+import lombok.SneakyThrows;
+
+public class FileHelper {
+
+    private FileHelper() {}
+
+    @SneakyThrows
```
Comment (martinmcclintock) [inline src/test/java/io/github/jpmorganchase/fusion/pact/util/FileHelper.java:11]: Why this?

File: src/test/resources/io/github/jpmorganchase/fusion/pact-samples/110274-fusionsdk-consumer-110274-fusionupload-provider.json
Comment (martinmcclintock) [inline src/test/resources/io/github/jpmorganchase/fusion/pact-samples/110274-fusionsdk-consumer-110274-fusionupload-provider.json:1]: What do we use this for?

File: src/test/resources/logback-test.xml
Comment (martinmcclintock) [inline src/test/resources/logback-test.xml:1]: Did you meant to commit these?

File: src/main/java/io/github/jpmorganchase/fusion/Fusion.java (hunk 1 above)
Comment (knighto82) [reply src/main/java/io/github/jpmorganchase/fusion/Fusion.java:614]: Will have a think when I am doing the build for upload/download.

File: src/test/java/io/github/jpmorganchase/fusion/pact/util/FileHelper.java (hunk 2 above)
Comment (knighto82) [reply src/test/java/io/github/jpmorganchase/fusion/pact/util/FileHelper.java:11]: Will change - there is actually a better way of reading from a stream that won't throw an exception.

File: src/test/resources/io/github/jpmorganchase/fusion/pact-samples/110274-fusionsdk-consumer-110274-fusionupload-provider.json
Comment (knighto82) [reply src/test/resources/io/github/jpmorganchase/fusion/pact-samples/110274-fusionsdk-consumer-110274-fusionupload-provider.json:1]: Used for testing on-prem.  Will delete once we have a proper build under our belt.

Issue created to remove -> #32

File: src/test/resources/logback-test.xml
Comment (knighto82) [reply src/test/resources/logback-test.xml:1]: Will correct.

**Review by martinmcclintock — APPROVED** [review]

**Pull request merged**